### PR TITLE
Rename aws_s3 to s3_object (and deprecate bucket creation/deleting)

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

The `aws_s3` module (as it's known today) is primarily for managing **objects** within S3.  While it provides minimal support for creating S3 buckets, the feature set is very limited.  Support for the advanced bucket management features is provided via the `s3_bucket` modules (such as managing encryption settings).

Because the name `aws_s3` often puts the module at the top of the list of modules, well away from the `s3_bucket` module, it can be difficult for folks to discover the `s3_bucket` module leading them to assume that we simply have no support for the more complex `s3_bucket` management features.

As such, I suggest renaming the module to `s3_object` to make the intended scope more obvious and to improve the discoverability of `s3_bucket`.  At this time I **do not** recommend setting a deprecation date for the alias, the cost of an alias is minimal and we've had a lot of churn recently.

Additionally, deprecates the duplicated (but very limited) bucket creation/deletion functionality of `aws_s3`/`s3_object`

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- aws_s3
(s3_object)

##### ADDITIONAL INFORMATION

See for example #866 where there was an attempt to create duplicate functionality.

### CODE REVIEW

**Comment by alinabuzachis** [comment]
I totally agree, s3_object is more indicative of the module's functionality.

File: changelogs/fragments/859-s3_object.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+minor_changes:
+- aws_s3 - The ``aws_s3`` module has been renamed to ``s3_object``
+  (https://github.com/ansible-collections/amazon.aws/pull/869).
+deprecated_features:
+- s3_object - Support for creation and deletion of S3 buckets has been deprecated.  Please use the
```
Comment (alinabuzachis) [inline changelogs/fragments/859-s3_object.yml:5]: Doesn't need a redirect in meta/runtime as well?

Comment (tremble) [reply changelogs/fragments/859-s3_object.yml:5]: Yup, I forgot to git add it :)

File: plugins/modules/s3_object.py [hunk 2]
```diff
@@ -191,39 +220,40 @@
     description:
       - Whether the bucket name should be validated to conform to AWS S3 naming rules.
       - On by default, this may be disabled for S3 backends that do not enforce these rules.
-      - See https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+      - See the Amazon documentation for more information about bucket nameing rules
```
Comment (jillr) [inline plugins/modules/s3_object.py:223]:
```suggestion
      - See the Amazon documentation for more information about bucket naming rules
```

**Review by jillr — APPROVED** [review]
One typo, otherwise LGTM

**Pull request merged**